torch/_dynamo/exc.py

```diff
@@ -96,6 +97,10 @@ def __init__(self, error_type: UserErrorType, msg):
         """
         super().__init__(msg)
         self.error_type = error_type
+        self.message = msg
+
+    def __str__(self):
+        return f"{self.error_type.name}: {self.message}"
```

Comment:
I'm not sure this is a good idea. The error type should be an internal classifier: it doesn't matter to the user what kind of error it is. At most, I'd be OK with the user seeing <SomeType>UserError, but never UserError(<SomeType). I'd also be OK with some numeric error code that doesn't detract from the error message and that we don't have to keep tweaking the name of as we refactor / rename error types.